### DISCUSSION TITLE
GPU acceleration error

### DISCUSSION THREAD

**Discussion by club660** [vincentjzy/OpenCorr#12] [Q&A]
Hi developers, 

I got following errors when I tried gpu acceleration with dic+fftcc+icgn2:

`Exception thrown at 0x00007FFCB8807C73 (OpenCorrGPULib.dll) in OpenCorr.exe: 0xC0000005: Access violation reading location 0x0000017B47DA801C.`

I have checked all the variables `gpu_ref_img, gpu_tar_img, gpu_POI_queue, icgn_conf` they all have values. 

And sometimes the error come with 

`CUDA error at d:\essential\opencorr\opencorrgpulib-20210322\opencorrgpulib\opencorr.cu:61 code=2(cudaErrorMemoryAllocation) "cudaHostAlloc((void**)&h_pois, bytes, 0x00)"`

The fact is my desktop don't have 'd' derive. Is that some hard code file paths in gpu library?

Kind regards
Zhengduo

**Comment by vincentjzy** [comment]
Hi Zhengduo, could you give more details about your GPU device, CUDA version and DIC prococessing parameters, like number of POIs and subset size, as well as the maximum displacement in your example? These may help us analyze the issue.

**Reply by club660** [reply]
Image display (github not support upload .bmp image file, these two are screenshot of images):
![image](https://user-images.githubusercontent.com/100261153/219294847-7c3dd044-b2c3-49a3-8c48-6885672c5201.png)
![image](https://user-images.githubusercontent.com/100261153/219295063-d4f37ae5-509c-4922-96f6-318351243c96.png)

Parameter info:
Image size 1001x801
radius size 15x15
max iteration  10
max deformation norm 0.001f
upper left point (16 16)
poi number 969x769
grid space 1

Device info:
Device count: 1
Using Device 0
DeiveName: NVIDIA RTX A6000
Memory: 49140MB
Cores: 84

CUDA info:
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Mon_Oct_11_22:11:21_Pacific_Daylight_Time_2021
Cuda compilation tools, release 11.4, V11.4.152
Build cuda_11.4.r11.4/compiler.30521435_0

I have tried with CPU version and no problem.

**Comment by vincentjzy** [comment]
It seems not the problem of ICGNGPU. Please check whether the initial guesses for ICGN are in a roughly acceptable range.

**Reply by club660** [reply]
It comes with same error even I use the samples you provided online. I will check on other desktop to see if this is general problem.

**Reply by vincentjzy** [reply]
The example of ICGNGPU was verified on a few desktops and laptops equipped with various graphics cards. You may try it using a geforce card.
